v3/memo.go

```diff
@@ -88,6 +87,51 @@ func (e *memoExpr) fingerprint() string {
 	return buf.String()
 }
 
+func (e *memoExpr) groupFingerprint() string {
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, "%s", e.op)
+
+	switch t := e.private.(type) {
+	case nil:
+	case *table:
+		fmt.Fprintf(&buf, " %s", t.name)
+	default:
+		fmt.Fprintf(&buf, " %s", e.private)
+	}
+
+	if f := e.physicalProps.fingerprint(); f != "" {
```

Comment:
How come the physical properties are part of memoExpr? I expected these to be in a separate hashtable in the memo group, as shown on page 342 of the Orca paper, something like this:
```
type memoGroup struct {
    physicalMap map[physicalProps]exprID
```

When a lower cost expression having a particular set of physical properties is found, it replaces the higher cost expression(s) in the memo group hash. Please note that my example here is very simplified, since there are other complications that might mean a different data structure, like support for property wildcard matching. See the Orca paper for examples.

Back to what that means for this method, two memo expressions with the same logical properties should be in the same memo group, even if they have different physical properties. But if you add the physical properties into the group fingerprint, then that would force the expressions into different memo groups, which seems wrong.

Reply:
There is no reason to include the physical properties in `groupFingerprint()`. That was copy&paste from `fingerprint()`. I don't think the physical properties need to be included in `fingerprint()` either, though it is nice for debugging purposes since the show up in `memo.String()`.

Reply:
Also note that `memoExpr.groupFingerprint()` is one of two mechanisms for grouping logically equivalent expressions. The other mechanism, which overrides `groupFingerprint()`, is that the root of a transformed expression is forced back into the same group as the originating expression.

Reply:
I just refreshed my memory about the Orca paper. The `memoGroup` structure is a map from physical properties to the best (lowest cost) expression providing those properties. I'm not sure where Orca stores the properties for an expression. I think they might be hung off `COperator` which in Orca is an class. Yep. `CPhysicalSort` is initialized with the required sort order.